### PR TITLE
Added PaystackSubscriptions class, methods, documentation and tests, fixed failing spec & changed minimum ruby and rest-client versions

### PR DESCRIPTION
**Pull request by timigod**

### CODE REVIEW

File: lib/paystack.rb [hunk 1]
```diff
@@ -5,50 +5,51 @@
...
+      raise PaystackBadKeyError, "No private key supplied and couldn't find any in environment variables. Make sure to set private key as an environment variable PAYSTACK_PRIVATE_KEY"
+    end
+    unless @private_key[0..2] == 'sk_'
+      raise PaystackBadKeyError, "Invalid private key #{@private_key}"
+    end
+
+
+  end
+
+  #TODO delete if not used
```
Comment (IkoroVictor) [inline lib/paystack.rb]: We don't need a setter for keys and even if we  needed a setter, it has to go through some kind of validation logic...here it just "sets" it...Its best the Paystack object is immutable and values passed only via constructor

File: spec/paystack_subscriptions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require 'spec_helper'
+require 'paystack/objects/subscriptions.rb'
+require 'paystack.rb'
+
+
+describe PaystackSubscriptions do
```
Comment (IkoroVictor) [inline spec/paystack_subscriptions_spec.rb]: This basically does'nt test anything... expect true = true????

File: paystack.gemspec [hunk 3]
```diff
@@ -25,6 +25,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "rspec", "~> 0"
   
   #Dependencies
+  spec.required_ruby_version = '>= 2.0.0'
```
Comment (IkoroVictor) [inline paystack.gemspec]: ok...

File: paystack.gemspec [hunk 4]
```diff
@@ -26,6 +26,6 @@ Gem::Specification.new do |spec|
   
   #Dependencies
   spec.required_ruby_version = '>= 2.0.0'
-  spec.add_dependency "rest-client", '~> 1.8'
+  spec.add_dependency "rest-client", '>= 2.0'
```
Comment (IkoroVictor) [inline paystack.gemspec]: has it been tested on your local branch?

File: lib/paystack.rb (hunk 1 above)
Comment (timigod) [inline lib/paystack.rb]: This wasn't code I wrote. This was all stuff you wrote. All I did was update the minimum ruby and rest client versions required.

File: paystack.gemspec (hunk 4 above)
Comment (timigod) [inline paystack.gemspec]: Yes it has been tested.

File: lib/paystack.rb (hunk 1 above)
Comment (IkoroVictor) [inline lib/paystack.rb]: ok.sorry my bad....

Comment (IkoroVictor) [inline lib/paystack.rb]: could you please remove the setters..

**Comment by timigod** [comment]
👍🏽

**Pull request merged**